ui/tests/integration/components/kubernetes/page/overview-test.js

```diff
@@ -0,0 +1,97 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { setupEngine } from 'ember-engines/test-support';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { render } from '@ember/test-helpers';
+import { typeInSearch, clickTrigger, selectChoose } from 'ember-power-select/test-support/helpers';
+import hbs from 'htmlbars-inline-precompile';
+
+module('Integration | Component | kubernetes | Page::Overview', function (hooks) {
+  setupRenderingTest(hooks);
+  setupEngine(hooks, 'kubernetes');
+  setupMirage(hooks);
+
+  hooks.beforeEach(function () {
+    this.store = this.owner.lookup('service:store');
+    this.store.pushPayload('secret-engine', {
+      modelName: 'secret-engine',
+      data: {
+        accessor: 'kubernetes_f3400dee',
+        path: 'kubernetes-test/',
+        type: 'kubernetes',
+      },
+    });
+    this.store.pushPayload('kubernetes/config', {
+      modelName: 'kubernetes/config',
+      backend: 'kubernetes-test',
+      ...this.server.create('kubernetes-config'),
+    });
+    this.store.pushPayload('kubernetes/role', {
+      modelName: 'kubernetes/role',
+      backend: 'kubernetes-test',
+      ...this.server.create('kubernetes-role'),
+    });
+    this.store.pushPayload('kubernetes/role', {
+      modelName: 'kubernetes/role',
+      backend: 'kubernetes-test',
+      ...this.server.create('kubernetes-role'),
+    });
```

Comment:
You can use the kubernetes mirage handler if you would like by setting and unsetting it in the before and after hooks:
```js
hooks.before(function () {
  ENV['ember-cli-mirage'].handler = 'kubernetes';
});

hooks.after(function () {
  ENV['ember-cli-mirage'].handler = null;
});
```
This is more of an FYI and what you did here is perfectly fine. The default scenario is not executed in the test so you would have to seed the DB initially in the before hook:
```js
this.server.create('kubernetes-config');
this.server.createList('kubernetes-role', 2);
```
Totally up to you if you would like to explore that route but I just wanted to let you know that option exists for the future.

Reply:
Looking at this again today with a clear head I realize now that this component is not responsible for loading the models so I led you astray with this suggestion :disappointed:. Your original approach was best! Sorry for the run around. Keep this in mind for acceptance tests when the endpoints are getting hit.